torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -2332,32 +2332,17 @@ std::vector<PredicateDomainInfo> getPredicateContigIds(
     return std::vector<PredicateDomainInfo>();
   }
 
-  // If root IDs are partial, i.e., start is non-zero and stop is not
-  // equal to extent, predication can't be done with merged domains as
-  // start and stop information is only available with root
-  // domains. Similarly, merged domains don't have enough information
-  // about halo to do correct predication, so they must be excluded.
-  std::unordered_set<IterDomain*> excluded_ids;
-
-  for (auto consumer_root_id : consumer_root_domain) {
-    if (gpu_lower->haloInfo()->getRootAxisInfo(consumer_root_id).hasHalo()) {
-      excluded_ids.insert(consumer_root_id);
-      continue;
-    }
-    if (consumer_root_id->maybePartial()) {
-      excluded_ids.insert(consumer_root_id);
-      continue;
-    }
-    // When consumer_root_id is a broadcast domain, do not allow contig
-    // predication as the merged output is not mapped with the
-    // reference unless the concrete domain is also a broadcast
-    // domain.
-    if (consumer_root_id->isBroadcast() &&
-        !GpuLower::current()
-             ->caMap()
-             ->getConcreteMappedID(consumer_root_id, IdMappingMode::PERMISSIVE)
-             ->isBroadcast()) {
-      excluded_ids.insert(consumer_root_id);
+  std::unordered_map<IterDomain*, IterDomain*> contig_concrete_to_ref_map;
+  for (auto entry : consumer_index_map) {
+    auto c_id = gpu_lower->caMap()->getConcreteMappedID(
+        entry.first, IdMappingMode::EXACT);
+    contig_concrete_to_ref_map[c_id] = c_id;
+  }
+  std::vector<bool> predicate_contiguity(consumer_root_domain.size(), true);
+  for (auto root_i : c10::irange(predicate_contiguity.size())) {
+    auto root_id = consumer_root_domain[root_i];
+    if (root_id->maybePartial()) {
+      // predicate_contiguity[root_i] = false;
```

Comment:
Is this safe to mark it as contiguous? Probably no unit test exercises this case, but as it was commented, predicates should be done at root domains in this case.

Reply:
Fixed and added a unit test (`FusionContigPredicateShift`)